spec/library/set/shared/inspect.rb

```diff
@@ -7,9 +7,10 @@
     Set[:a, "b", Set[?c]].send(@method).should be_kind_of(String)
   end
 
-  it "correctly handles self-references" do
-    (set = Set[]) << set
-    set.send(@method).should be_kind_of(String)
-    set.send(@method).should include("#<Set: {...}>")
-  end
+  # FIXME: Uncommenting this test causes VariableGetInstruction#generate to choke.
+  # it "correctly handles self-references" do
+  #   (set = Set[]) << set
```

Comment:
It's just the line that creates the set that causes the error here. Instead of uncommenting the whole test, it's probably better to rewrite it to something Natalie can understand
```ruby
it "correctly handles self-references" do
  # NATFIXME: Support this syntax
  # (set = Set[]) << set
  set = Set[]
  set << set
```
I'm not sure if this test is going to pass, the self reference is not handled in this code, it might be handled by the Array class, but I wouldn't be surprised when this would cause an infinite loop. That means we still would have to disable the test, but with a bit more specific message.

A few more generic pointers:
* To disable a spec, it's easier to replace `it` by `xit` (or `describe` by `xdescribe` when you're disabling a group of specs). This makes the difference between the upstream spec and our copy smaller, it's easier to search for disabled spec just by searching for those x-words, and we could add a test to check if they are still failing
* We use `NATFIXME` instead of `FIXME` to mark things that have to be fixed in Natalie. There are occasional FIXME markers in the upstream specs, and it's useful to distinguish between upstream and Natalie issues.

Reply:
@herwinw, thank you for your detailed feedback. You were correct, even changing the self reference syntax still does cause the spec to fail; therefore, it's x'd out for now.